framework/include/executioners/EigenExecutioner.h

```diff
@@ -0,0 +1,46 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef EIGENEXECUTIONER_H
+#define EIGENEXECUTIONER_H
+
+#include "Steady.h"
+
+class InputParameters;
+class EigenExecutioner;
+class EigenProblem;
+
+template <typename T>
+InputParameters validParams();
+
+template <>
+InputParameters validParams<EigenExecutioner>();
+
+
+class EigenExecutioner : public Steady
```

Comment:
Before reviewing, I do not quite like the name. We will have something like
```
[Executioner]
  type = EigenExecutioner
  ...
[]
```
Executioner word appeared twice. Maybe it is better just use `Eigenvalue`?